src/TrackTraversingAlgorithms.cpp

```diff
@@ -0,0 +1,127 @@
+#include "TrackTraversingAlgorithms.h"
+#include "CPUSolver.h"
+#include "Quadrature.h"
+
+
+/**
+ * @brief Constructor for SegmentSplitter calls the TraverseTracks
+ *        constructor
+ * @param track_generator The TrackGenerator to pull tracking information from
+ */
+VolumeCalculator::VolumeCalculator(TrackGenerator* track_generator)
```

Comment:
Do you even need to specify this dummy constructor? Or is it implicitly defined to simply call the `TraverseTracks` constructor if you don't define it for `VolumeCalculator`? I think you can remove it altogether.